conda_build/build.py

```diff
@@ -1276,7 +1276,8 @@ def test(recipedir_or_package_or_metadata, config, move_broken=True):
 
     warn_on_use_of_SRC_DIR(metadata)
 
-    config.compute_build_id(metadata.name())
+    if config.set_build_id:
```

Comment:
So I checked and the work dir with ```--no-build-id``` was not being deleted because the [cleanup code](https://github.com/conda/conda-build/blob/2.1.15/conda_build/config.py#L441) was taking the work dir modified by the test phase even when there is no test. So this PR also fix this case.